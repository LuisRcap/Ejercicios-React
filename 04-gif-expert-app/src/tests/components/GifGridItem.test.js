import { render } from '@testing-library/react';
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas de <GifGridItem />', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe mostrar el componente correctamente', () => {
        const { container } = render( <GifGridItem title={title} url={url} />);
        expect( container ).toMatchSnapshot();
    });
    /* const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem
            title={title}
            url={url}
        /> );

    test('Debe mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {
       
        const img = wrapper.find('img');
        // console.log( img.prop('src') );

        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });
    
    test('Debe tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect( className.includes('animate__fadeIn') ).toBe( true );

    }); */
    
    
    
});
