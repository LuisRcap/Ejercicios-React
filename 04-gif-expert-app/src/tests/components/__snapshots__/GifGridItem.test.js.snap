// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
